ci-operator/step-registry/gcp/provision/private-dns-zone/gcp-provision-private-dns-zone-commands.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# save the exit code for junit xml file generated in step gather-must-gather
+# pre configuration steps before running installation, exit code 100 if failed,
+# save to install-pre-config-status.txt
+# post check steps after cluster installation, exit code 101 if failed,
+# save to install-post-check-status.txt
+EXIT_CODE=100
+trap 'if [[ "$?" == 0 ]]; then EXIT_CODE=0; fi; echo "${EXIT_CODE}" > "${SHARED_DIR}/install-pre-config-status.txt"; CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' EXIT TERM
+
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+GCP_PROJECT="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GCP_PROJECT}"
+fi
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+
+if [[ "${PRIVATE_ZONE_PROJECT_TYPE}" == "service-project" ]]; then
+  echo "$(date -u --rfc-3339=seconds) - The DNS private zone will be in service project (i.e. where the cluster resources are deployed). "
+  CLUSTER_PVTZ_PROJECT="${GCP_PROJECT}"
+elif [[ "${PRIVATE_ZONE_PROJECT_TYPE}" == "dedicated-dns-project" ]]; then
+  echo "$(date -u --rfc-3339=seconds) - The DNS private zone, along with the base domain, will be in the dedicated DNS project (i.e. another service project where DNS resources are configured). "
+  if [[ -f "${CLUSTER_PROFILE_DIR}/third_project_setting.json" ]]; then
+    echo "$(date -u --rfc-3339=seconds) - INFO: Reading third project settings from '${CLUSTER_PROFILE_DIR}/third_project_setting.json'..."
+    CLUSTER_PVTZ_PROJECT=$(jq -r '.project' "${CLUSTER_PROFILE_DIR}/third_project_setting.json")
+    GCP_BASE_DOMAIN=$(jq -r '.baseDomain' "${CLUSTER_PROFILE_DIR}/third_project_setting.json")
+  else
+    echo "$(date -u --rfc-3339=seconds) - INFO: '${CLUSTER_PROFILE_DIR}/third_project_setting.json' not found, using the default..."
+    CLUSTER_PVTZ_PROJECT="openshift-qe-regional"
+    GCP_BASE_DOMAIN="regional.qe.gcp.devcluster.openshift.com"
+  fi
+else
+  echo "$(date -u --rfc-3339=seconds) - ERROR: Unknown private zone project type '${PRIVATE_ZONE_PROJECT_TYPE}', abort. " && exit 1
+fi
+
+if [[ "${PRE_CREATE_PRIVATE_ZONE}" == "yes" ]]; then
```

Comment:
If this step is called, ideally users want to bring their own dns, I do not think we need another switch to control it.

Reply:
This env var controls whether to pre-create the private zone in the given project, as the [feature](https://issues.redhat.com/browse/OCPSTRAT-1182) supports either using the pre-existing zone or creating the zone if it does not exist (see below for example). 

**Requirements (aka. Acceptance Criteria):**
When deploying OCP on GCP XPN the Installer would use and create the DNS records in the existing zone in the third Service Project. It should attempt to create it if it does not exist.